Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Import necessary modules from the package
from titanic_survival_package.data_preprocessing import load_dataset,demonstrate_preprocessing, fill_missing_values, preprocess_data
from titanic_survival_package.feature_engineering import create_family_size_feature, extract_and_process_titles, add_interaction_terms, drop_unnecessary_features
from titanic_survival_package.hyperparameter_tuning import tune_hyperparameters_grid, tune_hyperparameters_random
from titanic_survival_package.models import ModelTrainer, ModelEvaluator, train_stacking_model, generate_submission

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import StandardScaler
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[1], line 2[0m
[1;32m      1[0m [38;5;66;03m# Import necessary modules from the package[39;00m
[0;32m----> 2[0m [38;5;28;01mfrom[39;00m [38;5;21;01mtitanic_survival_package[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdata_preprocessing[39;00m [38;5;28;01mimport[39;00m load_dataset,demonstrate_preprocessing, fill_missing_values, preprocess_data
[1;32m      3[0m [38;5;28;01mfrom[39;00m [38;5;21;01mtitanic_survival_package[39;00m[38;5;21;01m.[39;00m[38;5;21;01mfeature_engineering[39;00m [38;5;28;01mimport[39;00m create_family_size_feature, extract_and_process_titles, add_interaction_terms, drop_unnecessary_features
[1;32m      4[0m [38;5;28;01mfrom[39;00m [38;5;21;01mtitanic_survival_package[39;00m[38;5;21;01m.[39;00m[38;5;21;01mhyperparameter_tuning[39;00m [38;5;28;01mimport[39;00m tune_hyperparameters_grid, tune_hyperparameters_random

[0;31mModuleNotFoundError[0m: No module named 'titanic_survival_package'

